### ISSUE TITLE
[Question] Is libva supported like in official telegram?

### ISSUE THREAD

**Issue by userofryzen** [kotatogram/kotatogram-desktop#388]
I mean for hardware video accel. Like a workaround I have enabled the open video files on local player until it gets supported or something( or not because I don't mind if I have to open video on another windows)

**Comment by ilya-fedin** [comment]
Well, official telegram doesn't have support for libva either. It only has NVDEC, Windows & macOS APIs support and all of them are so unstable that disabled by default.

**Comment by userofryzen** [comment]
> Well, official telegram doesn't have support for libva either. It only has NVDEC, Windows & macOS APIs support and all of them are so unstable that disabled by default.

official telegram has support for libva since I have tested it with an amd card in Linux, the part that is disabled by default i dont remember if it's true but it's stable at least for me have been being stable a few weeks
![imagen](https://user-images.githubusercontent.com/33232977/187427371-821fb93e-820c-4317-88a3-bece4a278480.png)

and it's working(checked with VCN block enabled when playing a video)

**Comment by ilya-fedin** [comment]
magic

**Comment by userofryzen** [comment]
> magic

sorry if it sounded bad or soemthing. English is not my native language and I was only pointing that there is on the official client

**Comment by ilya-fedin** [comment]
Well, I pointed that it's unexpected

**Comment by userofryzen** [comment]
maybe in the future releases  is it possible @ilya-fedin ?

**Comment by ilya-fedin** [comment]
AFAIK tdesktop doesn't support libva and we don't have enough knowledge to integrate it to tdesktop codebase, so the answer is no

I don't know why it works for you though, this might be some addition of third party build if you use one or maybe some other magic.

**Comment by userofryzen** [comment]
> AFAIK tdesktop doesn't support libva and we don't have enough knowledge to integrate it to tdesktop codebase, so the answer is no
> 
> I don't know why it works for you though, this might be some addition of third party build if you use one or maybe some other magic.

I'm using the simple and official client from official repos of arch(manjaro) not anymore. Any weird thing since arch is not adding anything more than upstream to libs, apps and everything. 
So.. yeah, libva is supported on the official ones from the latest update a few weeks ago. 
But it's okey with that you don't know how to do it. Don't worry about because opening the video on celuloid do the job in a correct way so..
But srly, it's true, telegram supports video acceleration from the latests versions with amdgpu and intel at least

**Comment by ilya-fedin** [comment]
> I'm using the simple and official client from official repos of arch(manjaro) not anymore.

Official distribution repos are third party packages from Telegram point of view, they can build Telegram with other versions of libraries built with other configuration and even patch Telegram sources adding addition features (e.g. Debian is known to do that). Try the binary from https://desktop.telegram.org and see what happens.

**Comment by userofryzen** [comment]
> > I'm using the simple and official client from official repos of arch(manjaro) not anymore.
> 
> Official distribution repos are third party packages from Telegram point of view, they can build Telegram with other versions of libraries built with other configuration and even patch Telegram sources adding addition features (e.g. Debian is known to do that). Try the binary from https://desktop.telegram.org and see what happens.

tested. 
![imagen](https://user-images.githubusercontent.com/33232977/188255461-d274b81d-67be-4cd7-b730-5cee063bc3f2.png)
but it's not working at all because it tries to load libcuda and not libva, no sense way . 
But it's unabled by default, the only thing you have to do on arch /manjao package is to enable it and it works out of the box

**Comment by userofryzen** [comment]
$ telegram-desktop -- %u
error: : cannot open
error: : cannot open
error: : cannot open
QObject::~QObject: Timers cannot be stopped from another thread
QObject::~QObject: Timers cannot be stopped from another thread
$ /tmp/firefox/Telegram/Telegram -workdir /home/albertogm/.local/share/TelegramDesktop/ -- %u
Invalid return value 0 for stream protocol
Invalid return value 0 for stream protocol
[AVHWDeviceContext @ 0x7f6566ed9b40] Cannot load libcuda.so.1
[AVHWDeviceContext @ 0x7f6566ed9b40] Could not dynamically load CUDA
 These are the different outputs I have

**Comment by userofryzen** [comment]
on the first the accel is working and on the second one there is no libva support
But I have not tested flatpak nor snap. Yet

**Comment by ilya-fedin** [comment]
Yeah, that's the point, seem to be some Arch/Manjaro addition

**Comment by userofryzen** [comment]
I dont think so, I think it's a bug from the official site because unless this the app should not decode the video through gpu, it0s not that easy to implement that on a program xD
So I have reported this, let's see what answer me on the issue https://github.com/telegramdesktop/tdesktop/issues/25018
